### PR TITLE
[nn] Remove deprecation warning from nn.functional.{tanh, sigmoid}

### LINKED ISSUE #65909: Deprecation: Remove nn.functional.tanh
This operation is deprecated and warns whenever it's used, see it's implementation here:

https://github.com/pytorch/pytorch/blob/4666e3f19228ddb73294c3ce7eba01b1641c083a/torch/nn/functional.py#L1835

The deprecation warning also correctly points users to torch.tanh, for which nn.functional.tanh is an alias. We should finally remove this operator.

cc @albanD @jbschlosser @walterddr @kshitij12345 @saketh-are @mruberry

### PR DESCRIPTION
**Pull request by foram-chandra**
Fixes #65909

### CODE REVIEW

**Comment by foram-chandra** [comment]
As mentioned in the issue about keeping activation functions in one namespace, would it make sense to remove deprecation warning from nn.sigmoid ?
https://github.com/pytorch/pytorch/blob/80556f23d332e494ac1ffc5d940d752267d84bf3/torch/nn/functional.py#L1966

**Review by albanD — APPROVED** [review]
Thanks!

**Comment by albanD** [comment]
There might be a test for that. Well see in CI

**Comment by mruberry** [comment]
> @mruberry is there anything special that needs to be done for "aliasing"?

Yes, depending on the alias. In this particular case I would do the following:

```
def tanh(input, *, out=None):
    r"""
    Alias for torch.tanh.
    
    See :class:`~torch.nn.Tanh` for the corresponding module.
    """

    return torch.tanh(input, out=out)
```

It would be nice to just do

```
tanh = torch.tanh
```

But I don't think our doc system can easily handle that. 

I also don't think we expose an `nn::functional::tanh` in our C++ API, so that shouldn't be an issue.

**Comment by Atcold** [comment]
It's not `nn.tanh` but `nn.functional.tanh`.
Also, the same warning should be removed for all the activation functions in `nn.functional`.

**Comment by lezcano** [comment]
@foram-chandra what's the state of this PR?

**Comment by foram-chandra** [comment]
@mruberry I agree with @Atcold that these functional activations should not have `out` kwarg support.

> @foram-chandra what's the state of this PR?

@lezcano I will remove the warning from `nn.functional.sigmoid` and I hope this PR should be good to go.

**Review by albanD — APPROVED** [review]
SGTM

**Comment by albanD** [comment]
The test failure does look relevant though.

**Comment by Atcold** [comment]
> The test failure does look relevant though.

Then?

**Comment by albanD** [comment]
> Then?

It needs to be addressed before we can land the PR!

**Comment by Atcold** [comment]
> > Then?
> 
> It needs to be addressed before we can land the PR!

By?

**Comment by albanD** [comment]
> By?

The PR author

**Comment by albanD** [comment]
@foram-chandra if you don't know how to fix these, don't hesitate to ask!

**Comment by kit1980** [comment]
I don't see any test issues after rebase, merging this.

**Review by kit1980 — APPROVED** [review]

**Comment by Atcold** [comment]
Can we plz fix this?
I still get the warnings on the latest PyTorch version (1.13.1).
